### DISCUSSION TITLE
Support for Parallel Displays using GPIO > 31 for Data

### DISCUSSION THREAD

**Discussion by dkalliv** [Bodmer/TFT_eSPI#2277] [Compatible displays and setup files]
Hi Bodmer,
I recently picked up a "WT32-SC01 Plus" display, which adresses many of the original WT32-SC01's shortcomings and uses 8Bit parallel communication. Unfortunately, the manufacturer chose to wire TFT_D1 to GPIO46, which currently does not work with TFT_eSPI. Is there any way to support Data pins using GPIO > 31 (I understand, there will be a performance penalty)?
LovyanGFX does not seem to have this restriction, but I do prefer your library as it is much more mature and way "cleaner" (my perception ;-))

Thank you for any help you can provide

**Comment by Bodmer** [comment]
There are two incompatibilities:
1. The board uses an ST7701 controller which is not supported by this library.
2. The mix of high and low GPIO pins for the bus is not supported.

The only way around this would be to add a new ST7701 driver and hack the board to connect say the free GPIO3 to GPIO46 and then use GPIO3 as D1.

This situation may change in the future but not for quite some time...

**Reply by dkalliv** [reply]
Thanks for getting back to me. Actually the display uses a ST7796U controller which works quite nicely with the ST7796 driver. Maybe I should have mentioned, that I already performed some "micro surgery" on the board to get it working (moving D1 to GPIO10). Unfortunately this is not only a major pain but also voids all certifications, which would have been a major advantage versus manufacturing my own hardware.
If I understand you correctly, supporting a mix of high and low GPIO pins is not trivial and I should not feel bad, that I failed, when I tried a quick and dirty hack in TFT_eSPI_ESP32_S3.h (the board uses an ESP32-S3) ;-)

Oh and thank you for this great library! This is really great work!

**Reply by Bodmer** [reply]
OK, that is good. I assumed it was this board:
https://www.seeedstudio.com/WT32-3-5-Inch-Display-p-5542.html
The details cite a ST7701 controller, but it looks like that is wrong!

To give a good performance the library uses a single register write to load all 8 bits. Unfortunately the high and low GPIO banks are in different registers so this complicates matters.

**Reply by dkalliv** [reply]
Hi Bodmer,
In the meantime I have added support for the WT32-SC01 Plus module myself and created a pull request.
